### PR TITLE
[3.11]  Clarify re docs for byte pattern group names (GH-99308)

### PR DESCRIPTION
**Pull request by Kentzo**
* Issue: gh-99308

### CODE REVIEW

**Comment by slateny** [comment]
Per [comment](https://github.com/python/cpython/pull/99311#issuecomment-1364635414) I don't believe there should be a backport for this.

**Comment by Kentzo** [comment]
https://github.com/python/cpython/issues/99308#issuecomment-1373049346 says otherwise

File: Doc/library/re.rst [hunk 1]
```diff
@@ -418,7 +418,8 @@ The special characters are:
    +---------------------------------------+----------------------------------+
 
    .. deprecated:: 3.11
```
Comment (CAM-Gerlach) [inline Doc/library/re.rst:420]:
```suggestion
   .. deprecated-removed:: 3.11 3.12
```

Use the most appropriate and specific directive here

File: Doc/library/re.rst [hunk 2]
```diff
@@ -494,6 +495,8 @@ The special characters are:
...
```
Comment (CAM-Gerlach) [inline Doc/library/re.rst:496]:
```suggestion
   .. deprecated-removed:: 3.11 3.12
```

File: Doc/library/re.rst [hunk 3]
```diff
@@ -1015,7 +1018,8 @@ Functions
...
```
Comment (CAM-Gerlach) [inline Doc/library/re.rst:1019]:
```suggestion
   .. deprecated-removed:: 3.11 3.12
```

**Review by CAM-Gerlach — COMMENTED** [review]
Thanks @Kentzo . One set of suggestions, otherwise LGTM

**Review by hauntsaninja — APPROVED** [review]
Thanks for backporting this!

Looks like it's common enough for past versions of docs to contain "removed" for future versions of Python, so would you mind applying CAM's suggestions?

Edit: Since this has been lying around, I think this is fine to merge as is without the more specific directives; 3.12 isn't finalised and this seems like a short time between deprecation and removal.

**Pull request merged**